src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadsForQNamesFinder.java

```diff
@@ -0,0 +1,61 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.broadinstitute.hellbender.tools.spark.utils.HopscotchUniqueMultiMap;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import scala.Tuple2;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Find <intervalId,list<fastqread>> pairs for interesting template names.
+ */
+public final class ReadsForQNamesFinder {
+    private final HopscotchUniqueMultiMap<String, Integer, QNameAndInterval> qNamesMultiMap;
+    private final int nIntervals;
+    private final int nReadsPerInterval;
+    private final boolean includeMappingLocation;
+    private final boolean dumpFASTQs;
+
+    public ReadsForQNamesFinder( final HopscotchUniqueMultiMap<String, Integer, QNameAndInterval> qNamesMultiMap,
+                                 final int nIntervals, final boolean includeMappingLocation, final boolean dumpFASTQs ) {
+        this.qNamesMultiMap = qNamesMultiMap;
+        this.nIntervals = nIntervals;
+        this.nReadsPerInterval = 2 * qNamesMultiMap.size() / nIntervals;
+        this.includeMappingLocation = includeMappingLocation;
+        this.dumpFASTQs = dumpFASTQs;
+    }
+
+    public Iterable<Tuple2<Integer, List<SVFastqUtils.FastqRead>>> call( final Iterator<GATKRead> readsItr ) {
+        @SuppressWarnings({ "unchecked", "rawtypes" })
```

Comment:
No.  It's necessary because I'm downcasting from a raw type.  (Note that it's not possible in Java to write new List<>[nIntervals].  You can't make arrays of a generic type.  I had to write new List[nIntervals] instead.)